src/rt/sched/behaviourcore.h

```diff
@@ -893,29 +897,28 @@ namespace verona::rt
         Logging::cout() << "Release phase for behaviour " << bodies[i]
                         << Logging::endl;
       }
-      // FIXME: Shouldn't this be only over chains?
-      // Do we avoid the if this way?
-      for (size_t i = 0; i < cown_count; i++)
+
+      for (size_t i = 0; i < chain_count; i++)
       {
         yield();
-        auto slot = std::get<1>(cown_to_behaviour_slot_map[i]);
+        auto slot = chain_info[i].slot;
         Logging::cout() << "Setting slot " << slot << " to ready"
                         << Logging::endl;
-        if (slot->is_wait_2pl())
-          slot->set_ready();
+        // Is this ok?
```

Comment:
I think so.

I wonder if we should also be setting the READ_AVAILABLE here.  If we kept enough information at this point.